### PR TITLE
snowflake: 2.3.1 -> 2.4.0

### PR DESCRIPTION
**Pull request by marsam**
###### Description of changes

https://gitlab.torproject.org/tpo/anti-censorship/pluggable-transports/snowflake/-/raw/v2.4.0/ChangeLog
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/snowflake/default.nix [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
+    repo = "snowflake";
     rev = "v${version}";
-    sha256 = "sha256-4/ZTLyST73krOL87am28TM+1mktchpoCSaASMqQl5e8=";
+    sha256 = "sha256-Jr9ihExw++cRZpunP43DbDx0RTaGpgil0avs4mTW8tA=";
   };
 
-  vendorSha256 = "sha256-a2Ng+D1I0v5odChM6XVVnNwea/0SOTOmdm2dqKaSU3s=";
+  vendorHash = "sha256-66GqvwHPkMii5oXZV36ayYYkW1oaq5qTjkEA5BeS/5U=";
+
+  ldflags = [ "-s" "-w" ];
```
Comment (lourkeur) [inline pkgs/tools/networking/snowflake/default.nix:18]: Is there a reason to strip symbols now?

**Review by bbjubjub2494 — COMMENTED** [review]
:heavy_check_mark: Confirmed that the new repository is legitimate

:heavy_check_mark: Nixpkgs-review

File: pkgs/tools/networking/snowflake/default.nix [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
+    repo = "snowflake";
     rev = "v${version}";
-    sha256 = "sha256-4/ZTLyST73krOL87am28TM+1mktchpoCSaASMqQl5e8=";
+    sha256 = "sha256-DR1H5ncFPHZWQAwOZKkfRrjwfzhirSzwtvKesaRmqcA=";
   };
 
-  vendorSha256 = "sha256-a2Ng+D1I0v5odChM6XVVnNwea/0SOTOmdm2dqKaSU3s=";
+  vendorHash = "sha256-66GqvwHPkMii5oXZV36ayYYkW1oaq5qTjkEA5BeS/5U=";
+
+  ldflags = [ "-s" "-w" ];
```
Comment (lourkeur) [inline pkgs/tools/networking/snowflake/default.nix:18]:
```suggestion
```
That's fair but it's not agreed upon across nixpkgs so we have to keep consistency.

**Review by bbjubjub2494 — CHANGES_REQUESTED** [review]

File: pkgs/tools/networking/snowflake/default.nix (hunk 2 above)
Comment (marsam) [reply pkgs/tools/networking/snowflake/default.nix]: Sure, fixed!

**Review by bbjubjub2494 — APPROVED** [review]
Result of `nixpkgs-review pr 203923` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>onionshare</li>
    <li>onionshare-gui</li>
    <li>snowflake</li>
  </ul>
</details>

**Pull request merged**